engine/src/juliabox/plugins/bucket_gs/impl_gs.py

```
@@ -0,0 +1,113 @@
+            google_oauth = JBoxCfg.get("google_oauth")
+            creds = _ServiceAccountCredentials(
+                service_account_id=google_oauth["client_id"],
+                service_account_email=google_oauth["client_email"],
+                private_key_id=google_oauth["key"],
+                private_key_pkcs8_text=google_oauth["secret"],
+                scopes=[])
+            JBoxGS.CONN = build("storage", "v1", credentials=creds)
+        return JBoxGS.CONN
+
+    @staticmethod
+    def connect_bucket(bucket):
+        if bucket not in JBoxGS.BUCKETS:
+            JBoxGS.BUCKETS[bucket] = JBoxGS.connect().buckets().get(bucket=bucket).execute()
+        return JBoxGS.BUCKETS[bucket]
+
+    @staticmethod
+    def _get_mime_type(local_file):
+        mime = MimeTypes()
+        url = urllib.pathname2url(local_file)
+        mime_type = mime.guess_type(url)
+        return mime_type[0]
+
+    @staticmethod
+    def push(bucket, local_file, metadata=None):
+        objconn = JBoxGS.connect().objects()
+        fh = open(local_file, "rb")
+        media = MediaIoBaseUpload(fh, JBoxGS._get_mime_type(local_file))
+        if metadata:
+            retval = objconn.insert(bucket=bucket, media_body=media,
+                                    name=os.path.basename(local_file),
+                                    body={"metadata": metadata}).execute()
+        else:
+            retval = objconn.insert(bucket=bucket, media_body=media,
+                                    name=os.path.basename(local_file)).execute()
+        fh.close()
+        return retval
+
+    @staticmethod
+    def pull(bucket, local_file, metadata_only=False):
+        objname = os.path.basename(local_file)
+        if metadata_only:
+            k = None
+            try:
+                k = JBoxGS.connect().objects().get(bucket=bucket,
+                                                   object=objname).execute()
+            except:
+                pass
```

may be better not to silently ignore exception?